lib/Driver/ToolChains.cpp

```diff
@@ -1045,22 +1045,16 @@ static void getRuntimeLibraryPath(SmallVectorImpl<char> &runtimeLibPath,
                           getPlatformNameForTriple(TC.getTriple()));
 }
 
-static void getClangLibraryPathOnDarwin(SmallVectorImpl<char> &libPath,
-                                        const ArgList &args,
-                                        const ToolChain &TC) {
-  getRuntimeLibraryPath(libPath, args, TC);
-  // Remove platform name.
-  llvm::sys::path::remove_filename(libPath);
-  llvm::sys::path::append(libPath, "clang", "lib", "darwin");
-}
+static void getClangLibraryPath(const ToolChain &TC, const ArgList &Args,
+                                SmallString<128> &LibPath) {
```

Comment:
Nitpick: Please leave this as a SmallVectorImpl, so that callers can use any size SmallString.